### PR TITLE
Feat: Added feature for S3 triggers in local testing Storage

### PR DESCRIPTION
**Pull request by akshbhu**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by undefobj** [comment]
Will this require any documentation updates to local mocking? @yuth @kaustavghosh06

File: packages/amplify-util-mock/src/storage/storage.ts [hunk 1]
```diff
@@ -64,6 +67,77 @@ export class StorageTest {
...
+        backendPath,
+        "storage",
+        resourceName,
+        "s3-cloudformation-template.json"
+      );
+      const storageParams = context.amplify.readJsonFile(CFNFilePath);
+      const lambdaConfig = storageParams.Resources.S3Bucket.Properties.NotificationConfiguration.LambdaConfigurations;
+      if (lambdaConfig === undefined) {
+        return context.print.warning(
+          "S3Trigger has not yet been added to this bucket. Please add the trigger"
```
[the remark this answers is not in the document]
Comment (akshbhu) [reply packages/amplify-util-mock/src/storage/storage.ts]: removed comment for adding trigger

File: packages/amplify-util-mock/src/storage/index.ts [hunk 2]
```diff
@@ -20,7 +20,9 @@ export async function start(context) {
         }
         const testApi = new StorageTest();
```
[the remark this answers is not in the document]
Comment (akshbhu) [reply packages/amplify-util-mock/src/storage/index.ts]: changed to mock storage

File: packages/amplify-util-mock/src/storage/storage.ts [hunk 3]
```diff
@@ -64,6 +67,77 @@ export class StorageTest {
...
+        );
+      }
+
+      // loop over lambda config to check for trigger based on prefix
+
+      let triggerName;
+      for (let obj of lambdaConfig) {
+        let prefix_arr = obj.Filter;
+        if (prefix_arr === undefined) {
+          if (String(eventObj.Records[0].event.eventName).split(':')[0] === "ObjectRemoved" || String(eventObj.Records[0].event.eventName).split(':')[0] === "ObjectCreated") {
```
[the remark this answers is not in the document]
Comment (akshbhu) [reply packages/amplify-util-mock/src/storage/storage.ts]: chnaged it to eventName

**Pull request merged**